### ISSUE TITLE
1D transformations

### ISSUE THREAD

**Issue by Ivorforce** [Timorleiderman/tensorflow-wavelets#6]
Hi!
I'm fitting some (batched) 1D data in keras. I'm noticing that the [DWT / IDWT](https://github.com/Timorleiderman/tensorflow-wavelets/blob/main/src/tensorflow_wavelets/Layers/DWT.py) layers in this library are designed for 2 dimensional data. 
Would it be in scope to add a 1 dimensional implementation as well? I'm guessing using a height of 1 will do the trick all the same, but using a dedicated solution would probably result in better performance (unless TF can fold the difference out).

**Comment by Timorleiderman** [comment]
Good point! I'll start working on a dedicated 1D implementation right away. 

Using a height of 1 should work as a workaround, but a proper 1D version will likely be more efficient. I'll update here once I make progress!

**Comment by Timorleiderman** [comment]
take a look at this notebook, is this implementation is good?
https://github.com/Timorleiderman/tensorflow-wavelets/blob/main/getting_started/dwt1d.ipynb

**Comment by Ivorforce** [comment]
Hi! This was very fast, thank you!
I've tried testing this directly in my project. Unfortunately, it seems to fail for any channel counts except 1:

```python
def create_sig(channel_count: int):
    num_samples = 8
    x = np.linspace(0, 2 * np.pi, num_samples)
    signal = np.sin(5 * x).astype(np.float32)  # 5 Hz sine wave
    return np.repeat(signal[None, :, None], repeats=channel_count, axis=-1)

for i in range(1, 10):
    print(i) # Makes it to 2, then fails
    dwt = DWT.DWT1D(wavelet_name="db2")(tf.dtypes.cast(create_sig(i), tf.float32))
    idwt = DWT.IDWT1D(wavelet_name="db2")(tf.dtypes.cast(dwt, tf.float32))
```
Let me know if I'm doing anything wrong!

**Comment by Timorleiderman** [comment]
I added a ficx for the channels issue

**Comment by Ivorforce** [comment]
Looks good! Thanks for the fix.
I'm still getting an error for multi-channel input (`Gradients for grouped convolutions are not supported on CPU.`), but I'm not using batches in my training, so I just transposed to put my channels into the batch dimension and keep the channels dimension at length 1.
My outputs are a bit odd, but it does run and compile, so I'll just toy around with it for a while and see if it's a problem on my end. Thanks again!
Edit: Figured it out now, works great!
